perf/load/templates/h2upgrade.yaml

```diff
@@ -0,0 +1,34 @@
+{{- define "h2upgrade" }}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: {{ .serviceNamePrefix }}{{ .serviceName }}-h2upgrade
+spec:
+  configPatches:
+  - applyTo: CLUSTER
+    match:
+      context: {{ .ctx }}
+      cluster:
+        service: {{ .serviceNamePrefix }}{{ .serviceName }}.{{ .Namespace }}.svc.cluster.local
+    patch:
+      operation: MERGE
+      value: # cluster specification
+        http2_protocol_options: { max_concurrent_streams: 47707 }
+---
+{{- end }}
+
+{{- if .Values.h2upgrade }}
+
+{{- range $i, $svcName := $.Values.services }}
+
+{{- $svc := dict "serviceName" $svcName "serviceNamePrefix" $.Values.serviceNamePrefix "Namespace" $.Values.Namespace }}
```

Comment:
this is going to have a fairly large pilot perf impact most likely (which is fine, just a heads up). Unless the CDS updates are more efficient than LDS; for LDS just adding the stats filter used 15% of pilot CPU